ioread.cpp

```diff
@@ -936,15 +1019,12 @@ template<unsigned long int N> bool readFsGridVariable(
                         + (y - thatTasksStart[1]) * thatTasksSize[0]
                         + (x - thatTasksStart[0]);
 
-                     memcpy(targetGrid.get(x - localStart[0], y - localStart[1], z - localStart[2]), &buffer[index*N], N*sizeof(Real));
+                     memcpy(targetGrid.get(x - localStart[0], y - localStart[1], z - localStart[2]), &(vectorOfBuffers.at(task).at(index*N)), N*sizeof(Real));
                   }
                }
             }
-         } else {
-            // If we don't overlap, just perform a dummy read.
-            file.endMultiread(fileOffset);
          }
-         fileOffset += thatTasksSize[0] * thatTasksSize[1] * thatTasksSize[2];
```

Comment:
See? We *were* doing something to that fileOffset, we don't any more. But hmm we should start at fileOffset 0 and use the manual offsets I suppose. If I hadn't botched this, that is...